test/Microsoft.AspNet.Mvc.FunctionalTests/ViewEngineTests.cs

```diff
@@ -289,5 +305,80 @@ public static IEnumerable<object[]> RazorViewEngine_UsesExpandersForLayoutsData
             // Assert
             Assert.Equal(expected, body.Trim());
         }
+
+        [Fact]
+        public async Task ViewComponentsExecuteLayout()
+        {
+            // Arrange
+            var expected =
+@"<title>View With Component With Layout</title>
+
+Page Content
+<component-title>ViewComponent With Title</component-title>
+<component-body>
+Component With Layout</component-body>";
+            var server = TestServer.Create(_provider, _app);
+            var client = server.CreateClient();
+
+            // Act
+            var body = await client.GetStringAsync("http://localhost/ViewEngine/ViewWithComponentThatHasLayout");
+
+            // Assert
+            Assert.Equal(expected, body.Trim());
+        }
+
+        [Fact]
+        public async Task PartialDoNotExecuteViewStarts()
+        {
+            // Arrange
+            var expected = "Partial that does not specify Layout";
+            var server = TestServer.Create(_provider, _app);
+            var client = server.CreateClient();
+
+            // Act
+            var body = await client.GetStringAsync("http://localhost/PartialsWithLayout/PartialDoesNotExecuteViewStarts");
+
+            // Assert
+            Assert.Equal(expected, body.Trim());
```

Comment:
@pranavkm does the above make sense? How does the view engine know **not** to include view starts when rendering a partial/view-component view?